gubernator/gcs_async.py

```diff
@@ -26,7 +28,19 @@
 @ndb.tasklet
 def get(url):
     context = ndb.get_context()
-    headers = {'accept-encoding': 'gzip, *', 'x-goog-api-version': '2'}
+
+    headers = {
+        'accept-encoding': 'gzip, *',
+        'x-goog-api-version': '2',
+        }
+
+    url_result = urlparse.urlparse(url)
+    if url_result.netloc.endswith('.googleapis.com'):
```

Comment:
I don't know how to make the stub happy with this. You could guard it all with `if 'testbed' not in os.environ.get('SERVER_SOFTWARE', 'testbed'):`.

Gross :(

Reply:
see my previous comment, I think `install_handler` need to be explicitly called for `listdirs`?